src/main/java/org/opentripplanner/model/plan/pagecursor/PageCursor.java

```diff
@@ -4,39 +4,45 @@
 import java.time.Instant;
 import javax.annotation.Nullable;
 import org.opentripplanner.model.base.ToStringBuilder;
+import org.opentripplanner.model.plan.SortOrder;
 
 /**
  * This class hold all the information needed to page to the next/previous page. It is
  * serialized as base64 when passed on to the client. The base64 encoding is done to prevent the
  * client from using the information inside the cursor.
  * <p>
- * This class is internal to the router, only the serialized string is passed to/from the clients.
+ * The PageCursor class is internal to the router, only the serialized string is passed to/from the
+ * clients.
  */
 public class PageCursor {
+    public final PageType type;
+    public final SortOrder originalSortOrder;
```

Comment:
Isn't this deductible from the routing request?

Reply:
I changed the paging to switch from `arrive-by` to `depart-after` when paging next. So the `arriveBy` flag in the request have the original value, witch make the logic very complicated. E.g. for a arrive-by(true) search we page: **previous(arriveBy=true)**, then next(false), then **previous(false)**. We set the `sortOrder` and not the arriveBy flag in the cursor, because this make it explicit and enable us to support more than two ways to sort in the future.

As you pointed out - the code was a bit hard to understand (reverseFilteringDirection). It becomes easier to understand if we store the state(pageType, sortOrder) of the system and not the needed action to take (reverseFilteringDirection). In this case the information is needed in the RoutingWorker and in the PageCursorFactory in addition to the ItineraryFilter. A good design principle is to keep enough information for other components to take their own decisions, instead of passing a flag telling them what to do.

Reply:
👍